File: src/main/java/com/bookstore/booksmanagementsystem/service/ClassService.java
package com.bookstore.booksmanagementsystem.service;

import com.bookstore.booksmanagementsystem.dto.ClassDTO;

import java.util.List;

public interface ClassService {
	ClassDTO createClass(ClassDTO classDTO);

	ClassDTO getClassById(Long id);

	List<ClassDTO> getAllClasses();

	ClassDTO updateClass(Long id, ClassDTO classDTO);

	void deleteClass(Long id);

	List<ClassDTO> getClassesBySchool(Long schoolId);
}

File: bookstore-frontend/src/services/classService.js
import api from './api';

const classService = {
  getAllClasses: () => api.get('/classes'),
  getClassById: (id) => api.get(`/classes/${id}`),
  createClass: (classData) => api.post('/classes', classData),
  updateClass: (id, classData) => api.put(`/classes/${id}`, classData),
  deleteClass: (id) => api.delete(`/classes/${id}`),
};

export default classService;
